src/components/Item/VisualizationItem/ItemHeaderButtons.js

```diff
@@ -15,178 +20,127 @@ import {
-    [REPORT_TABLE, EVENT_REPORT].includes(activeType)
-        ? activeStyle
-        : inactiveStyle(disabled);
-
-const chartBtnStyle = (activeType, disabled) =>
-    [CHART, EVENT_CHART].includes(activeType)
-        ? activeStyle
-        : inactiveStyle(disabled);
-
-const mapBtnStyle = (activeType, disabled) =>
-    [MAP].includes(activeType) ? activeStyle : inactiveStyle(disabled);
-
-class VisualizationItemHeaderButtons extends Component {
-    renderInterpretationButton() {
-        const { activeFooter, onToggleFooter } = this.props;
-
-        const toggleFooterBase = activeFooter ? activeStyle : baseStyle;
-
-        const toggleFooter = {
-            ...toggleFooterBase,
-            container: {
-                ...toggleFooterBase.container,
-                ...style.toggleFooterPadding,
-                ...style.border,
-            },
-        };
-
-        return (
-            <Fragment>
-                <ItemHeaderButton
-                    style={toggleFooter.container}
-                    onClick={onToggleFooter}
-                >
-                    <MessageIcon style={toggleFooter.icon} />
-                </ItemHeaderButton>
-            </Fragment>
-        );
-    }
-
-    renderVisualizationButtons() {
-        const {
-            item,
-            visualization,
-            onSelectActiveType,
-            activeType,
-        } = this.props;
-
-        if (isSingleValue(visualization.type)) {
-            return null;
-        }
+    const onViewTable = () => {
+        onSelectActiveType(isTrackerType ? EVENT_REPORT : REPORT_TABLE);
+        handleClose();
+    };
 
-        const onViewTable = () =>
-            onSelectActiveType(
-                isTrackerDomainType(item.type) ? EVENT_REPORT : REPORT_TABLE
-            );
+    const onViewChart = () => {
+        onSelectActiveType(isTrackerType ? EVENT_CHART : CHART);
+        handleClose();
+    };
 
-        const onViewChart = () =>
-            onSelectActiveType(
-                isTrackerDomainType(item.type) ? EVENT_CHART : CHART
-            );
+    const onViewMap = () => {
+        onSelectActiveType(MAP);
+        handleClose();
+    };
 
-        const onViewMap = () => onSelectActiveType(MAP);
+    const itemHasMapView = () => hasMapView(item.type);
 
-        // disable toggle buttons
-        let disabled = false;
+    const handleMenuClick = (_, event) => setAnchorEl(event.currentTarget);
 
-        if (item.type === CHART) {
-            if (extractFavorite(item).type.match(/^YEAR_OVER_YEAR/)) {
-                disabled = true;
-            }
+    const handleInterpretationClick = () => {
+        props.onToggleFooter();
+        if (anchorEl !== null) {
+            handleClose();
         }
+    };
+
+    const handleClose = () => setAnchorEl(null);
+
+    const type = visualization.type || item.type;
+    const canViewAs =
+        !isSingleValue(props.visualization.type) &&
+        !type.match(/^YEAR_OVER_YEAR/);
```

Comment:
Could we use the Analytics fn `isYearOverYear` instead?

Reply:
absolutely. Sometimes when moving code, you kind of miss those things :)